JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -4,40 +4,65 @@ Given 2 strings write a function that will check if it is
 1 or 0 edits away*/
 
 const isOneAway = (str1, str2) => {
+  const diffInLen = Math.abs(str1.length - str2.length);
+  if (diffInLen > 1) {
+    return false;
+  }
+
+  let errorCount = 0;
   if (str1.length === str2.length) {
-    let errorCount = 0;
     for (let i = 0; i < str1.length; i++) {
       if (str1[i] !== str2[i]) {
         errorCount++;
-      }
-      if (errorCount > 1) {
-        return false;
+        if (errorCount > 1) {
+          return false;
+        }
       }
     }
   } else {
-    let errorCount = 0;
-    const longestStr = findLongestStr(str1, str2);
-    let x = 0;
-    for(let i = 0; i < longestStr; i++) {
-      if (str1[i] !== str2[x]) {
+    const longStr = str1.length > str2.length ? str1 : str2;
+    const shortStr = str1.length < str2.length ? str1 : str2;
+    for (
+      let i = 0, x = 0;
+      i < longStr.length && x < shortStr.length;
+      i++, x++
+    ) {
+      if (longStr[i] !== shortStr[x]) {
```

Comment:
more opportunities to use the guard clause pattern!

```javascript
if (longStr[i] === shortStr[x]) {
  continue;
}
```

Reply:
Yes, TY. I am working on being more aware of this pattern